server/api/crud/pipelines.py

```diff
@@ -433,3 +440,23 @@ def _get_experiment_id_from_run(run: dict) -> str:
             ):
                 return data.get("id", "")
         return ""
+
+    def _filter_runs_by_name(self, runs: list, target_name: str) -> list:
+        """Filter runs by their name while ignoring the project string on them
+
+        :param runs: list of runs to be filtered
+        :param target_name: string that should be part of a valid run name
+        :return: filtered list of runs
+        """
+        if not target_name:
+            return runs
+
+        def filter_by(run):
+            run_name = run.get("name", "").replace(
```

Comment:
Using `replace` is too permissive - it will remove project name from anywhere in the string, not just the prefix. Should be replaced with something using `startswith` or since we're in py3.9, you can use `str.removeprefix()`.